serverManager/service/source/ConfigReader.cpp

```diff
@@ -75,33 +88,71 @@ void ConfigReader::parseEnvironmentVariables(std::shared_ptr<firebolt::rialto::w
     }
 }
 
+void ConfigReader::parseExtraEnvVariables(std::shared_ptr<firebolt::rialto::wrappers::IJsonValueWrapper> root)
+{
+    if (root->isMember("extraEnvVariables") && root->at("extraEnvVariables")->isArray())
+    {
+        std::shared_ptr<firebolt::rialto::wrappers::IJsonValueWrapper> envVarsJson = root->at("extraEnvVariables");
+        Json::ArrayIndex size = envVarsJson->size();
+        for (Json::ArrayIndex index = 0; index < size; ++index)
+        {
+            if (envVarsJson->at(index)->isString())
+            {
+                m_extraEnvVars.emplace_back(envVarsJson->at(index)->asString());
+            }
+        }
+    }
+}
+
 void ConfigReader::parseSessionServerPath(std::shared_ptr<firebolt::rialto::wrappers::IJsonValueWrapper> root)
 {
-    if (root->isMember("session_server_path") && root->at("session_server_path")->isString())
+    if (root->isMember("sessionServerPath") && root->at("sessionServerPath")->isString())
```

Comment:
This logic is repeated across multiple parsing methods, but with different key names, value types and variables, so maybe we could create a template helper to reduce the duplicated code making it cleaner - the reason I thought of a template is for type safety - but this might be unnecessary & an over optimisation in this case, lmk what you think

Reply:
done